### PR TITLE
Automate BZ 1389543

### PR DESCRIPTION
**Pull request by svtkachenko**
https://bugzilla.redhat.com/show_bug.cgi?id=1389543

```python
pytestrunner.py -p pytest_teamcity tests/foreman/api/test_product.py "-k ProductTestCase and test_positive_sync"
Testing started at 8:12 PM ...
2017-02-21 20:12:01 - conftest - DEBUG - Registering custom pytest_namespace

============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.14, cov-2.3.1
collected 16 items
====== 15 tests deselected by '-k ProductTestCase and test_positive_sync' ======
=================== 1 passed, 15 deselected in 37.80 seconds ===================

### CODE REVIEW

**Comment by oshtaier** [comment]
As far as I see, we don't have coverage for `product.sync()` at all, so probably you may add one more case, just basic one?:
```
product = entities.Product().create()
rpm_repo = entities.Repository(
    product=product,
    content_type='yum',
    url=FAKE_1_YUM_REPO
).create()
product.sync()
self.assertGreaterEqual(rpm_repo.read().content_counts['rpm'], 1)
```

**Comment by svtkachenko** [comment]
Additional test and asserts added.

```python
% py.test tests/foreman/api/test_product.py -k "sync"
===================================================================== test session starts =====================================================================

tests/foreman/api/test_product.py ..

=============================================================== 15 tests deselected by '-ksync' ===============================================================
========================================================== 2 passed, 15 deselected in 80.63 seconds ===========================================================
```

File: tests/foreman/api/test_product.py [hunk 1]
```diff
@@ -295,3 +300,51 @@ def test_positive_delete(self):
                 product.delete()
                 with self.assertRaises(HTTPError):
                     entities.Product(id=product.id).read()
+
+    @tier1
+    def test_positive_sync(self):
+        """Sync repository for a product
```
Comment (oshtaier) [inline tests/foreman/api/test_product.py]: Better to say that we sync product as we calling API for Product entity in that case

File: tests/foreman/api/test_product.py [hunk 2]
```diff
@@ -295,3 +300,51 @@ def test_positive_delete(self):
...
+            content_type='yum',
+            url=FAKE_1_YUM_REPO
+        ).create()
+        self.assertEqual(rpm_repo.read().content_counts['rpm'], 0)
+        product.sync()
+        self.assertGreaterEqual(rpm_repo.read().content_counts['rpm'], 1)
+
+    @tier2
+    def test_positive_sync_several_repos(self):
+        """Sync all repositories for a product
```
Comment (oshtaier) [inline tests/foreman/api/test_product.py]: Same there

**Comment by oshtaier** [comment]
ACK pending small nitpicks

File: tests/foreman/api/test_product.py (hunk 1 above)
Comment (svtkachenko) [inline tests/foreman/api/test_product.py]: However documentation for that API path says 'Sync all repositories for a product' :)

**Pull request merged**